Tests/pens/pointPen_test.py

```diff
@@ -0,0 +1,408 @@
+from __future__ import print_function, division, absolute_import
+from fontTools.misc.py23 import *
+from fontTools.misc.loggingTools import CapturingLogHandler
+import unittest
+
+from fontTools.pens.basePen import AbstractPen
+from fontTools.pens.pointPen import AbstractPointPen, PointToSegmentPen, \
+    SegmentToPointPen, GuessSmoothPointPen, ReverseContourPointPen
+
+
+class _TestSegmentPen(AbstractPen):
+
+    def __init__(self):
+        self._commands = []
+
+    def __repr__(self):
+        return " ".join(self._commands)
+
+    def moveTo(self, pt):
+        self._commands.append("%s %s moveto" % (pt[0], pt[1]))
+
+    def lineTo(self, pt):
+        self._commands.append("%s %s lineto" % (pt[0], pt[1]))
+
+    def curveTo(self, *pts):
+        pts = ["%s %s" % pt for pt in pts]
+        self._commands.append("%s curveto" % " ".join(pts))
+
+    def qCurveTo(self, *pts):
+        pts = ["%s %s" % pt if pt is not None else "None" for pt in pts]
+        self._commands.append("%s qcurveto" % " ".join(pts))
+
+    def closePath(self):
+        self._commands.append("closepath")
+
+    def endPath(self):
+        self._commands.append("endpath")
+
+    def addComponent(self, glyphName, transformation):
+        self._commands.append("%r %s addcomponent" % (glyphName, transformation))
+
+
+def _reprKwargs(kwargs):
+    items = []
+    for key in sorted(kwargs):
+        items.append("%s=%r" % (key, kwargs[key]))
+    return items
+
+
+class _TestPointPen(AbstractPointPen):
+
+    def __init__(self):
+        self._commands = []
+
+    def __repr__(self):
+        return " ".join(self._commands)
+
+    def beginPath(self, identifier=None, **kwargs):
+        items = []
+        if identifier is not None:
+            items.append("identifier=%r" % identifier)
+        items.extend(_reprKwargs(kwargs))
+        self._commands.append("beginPath(%s)" % ", ".join(items))
+
+    def addPoint(self, pt, segmentType=None, smooth=False, name=None,
+                 identifier=None, **kwargs):
+        items = ["%s" % (pt,)]
+        if segmentType is not None:
+            items.append("segmentType=%r" % segmentType)
```

Comment:
here, don't use `%r`, use just `%s`. the problem is the repr() of unicode objects in py27.

Reply:
By not using the unicode_literals directive in pointPen.py, the py27 tests seem to pass here locally.

Reply:
Not sure whether there may be other consequences of not using `unicode_literals` in pointPen.py. If you want I can use `'%s'` instead of `%r`. (Except the one in `_reprKwargs()`, but that's currently not even properly used/tested.)

Reply:
hm, I would actually prefer this latter option. I like to have all text strings as unicode (more future proof)